sharktank/sharktank/layers/paged_attention.py

```diff
@@ -26,9 +26,78 @@
     StaticScaledQuantizer,
 )
 from sharktank import ops, kernels
+from sharktank.kernels.mlir_kernel import *
 
 __all__ = ["PagedAttention"]
 
+# Paged Attention Kernels
+#
+# Each kernel is put into its own class to create a namespace for it
+
+
+def KVCacheGatherKernel():
```

Comment:
Either move the KV cache  gather to the ops implementation (or I can do that in a follow up). This should let us port it in and support sharded models as well.